projects/lights_control/test/test_lights_can.c

```diff
@@ -0,0 +1,109 @@
+#include "can.h"
+#include "event_queue.h"
+#include "interrupt.h"
+#include "status.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#include "lights_can.h"
+#include "lights_events.h"
+#include "lights_gpio.h"
+
+// test for initializing the CAN settings
+
+static CANMessageID s_msg_id = 0x1;
+
+void setup_test(void) {
+  gpio_init();
+  interrupt_init();
+  soft_timer_init();
+  event_queue_init();
+}
+
+void teardown_test(void) {}
+
+// static void prv_clock_tx(Event* e) {
+//   StatusCode ret;
+//   do {
+//     ret = event_process(e);
+//   } while (ret != STATUS_CODE_OK);
+//   TEST_ASSERT_EQUAL(EVENT_CAN_TX, e->id);
+//   TEST_ASSERT_TRUE(fsm_process_event(CAN_FSM, e));
+//
+//   do {
+//     ret = event_process(e);
+//   } while (ret != STATUS_CODE_OK);
+//   TEST_ASSERT_EQUAL(EVENT_CAN_RX, e->id);
+//   TEST_ASSERT_TRUE(fsm_process_event(CAN_FSM, e));
+// }
+
+void test_lights_rx_front(void) {
+  lights_can_init(LIGHTS_BOARD_FRONT, true);
+  CANMessage msg = { .msg_id = 0x1, .type = CAN_MSG_TYPE_DATA, .data = 0x4, .dlc = 2 };
+  uint16_t test_messages[] = { 0x0, 0x101, 0x002, 0x103, 0x004, 0x107 };
```

Comment:
Magic numbers? Where are these values coming from?

Reply:
magic numbers :P I just wanted them to have random values so that they're not "accidentally" asserted correctly.

Reply:
any suggestions on how to fix this? I'm basically asserting that the same data is received as sent, but I don't know how to do it in a more legitimate way. lol

Reply:
Use a loop to generate the messages for some range you know to be bad, or pick obviously bad ids such as `UINT16_MAX`.

Reply:
> I'm basically asserting that the same data is received as sent

Out of curiosity, why do you need to do that? It sounds like you're testing that CAN loopback works, which isn't something that you should need to do.

Reply:
How can I check for if an event should **not** be raised? 
Like here what I do is that I repeatedly call event_process till an event comes up. But if I pass in an invalid id, then  my can_rx_handler_front will return `STATUS_CODE_INVALID_ARGS`, and no event will be raised as a result. If I do that then this will keep looping... Unless I pass another (valid) message after, and assert that the corresponding event gets raised?

Reply:
> Out of curiosity, why do you need to do that? It sounds like you're testing that CAN loopback works, which isn't something that you should need to do.
Well my intention was to check whether the event is getting raised, with the correct data. I intended for this module to do that. Which is why I'm sending messages to it, and seeing if it raises events.